pkg/hostpath/hostpath.go

```diff
@@ -564,27 +627,33 @@ func deleteIfEmpty(name string) {
 	}
 }
 
+func (hp *hostPath) innerDeleteHostpathVolume(top string) {
+	// reminder, path is filepath.Join(DataRoot, [anchor-dir | bind-dir], volID, podNamespace, podName, podUID, podSA)
+	// delete SA dir
+	err := os.RemoveAll(top)
+	if err != nil {
+		klog.Warningf("error deleting %s: %s", top, err.Error())
+	}
+	uidPath := filepath.Dir(top)
+	deleteIfEmpty(uidPath)
```

Comment:
```suggestion
	deleteIfEmpty(filepath.Dir(top))
```

I don't think that you need to create a variable first and then use that in the `deleteIfEmpty` function, same with the below ones

Reply:
It is a little trickier than that, as each parameter for deleteIfEmpty builds off the last invocation, as we are working back through the file system directory tree.

That said, I'll have an alternative up that is in the spirit of what you are asking.  Let me know what you think once I post it.